### PR TITLE
Users and importing them and speed

### PR DESCRIPTION
**Pull request by csaunders**
This adds the users and stuff and some caching/fetching/storing whatever. Here's some performance numbers

Login with no friends in DB:
`Completed 200 OK in 3025ms (Views: 34.6ms | ActiveRecord: 190.2ms)`

Login with some friends in DB:
`Completed 200 OK in 2506ms (Views: 22.7ms | ActiveRecord: 52.2ms)`

Login with all friends in DB:
`Completed 200 OK in 1223ms (Views: 23.8ms | ActiveRecord: 3.5ms)`

Login with all friends and only calling Steam to get Users games list:
`Completed 200 OK in 935ms (Views: 27.9ms | ActiveRecord: 1.7ms)`

Please review @carsonb @davefp @dan-menard

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 
   def ensure_logged_in
     redirect_to(login_path) unless logged_in?
   end
 
+  def current_user
+    @current_user ||= User.find(session[:current_user]) if session[:current_user]
+  end
+
   def user_id
```
Comment (davefp) [inline app/controllers/application_controller.rb:21]: Should this not just be a method on User now?

File: app/controllers/auth_controller.rb [hunk 2]
```diff
@@ -7,11 +7,11 @@ def index
 
   def callback
     auth = request.env['omniauth.auth']
-    session[:current_user] = {
-      :nickname => auth.info['nickname'],
-      :image => auth.info['image'],
-      :uid => auth.uid
-    }
+
+    user = User.where(uid: auth.uid.to_i).first_or_initialize
```
Comment (davefp) [inline app/controllers/auth_controller.rb:11]: Does this set the uid when there's no user already?

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+class User < ActiveRecord::Base
+  serialize :friends, Array
+
+  # This is the caca and should be fixed
```
Comment (davefp) [inline app/models/user.rb]: Fix pls :wink:

File: app/controllers/auth_controller.rb (hunk 2 above)
Comment (csaunders) [inline app/controllers/auth_controller.rb:11]: First or initialize ftw
- Chris Saunders / sndrs.ca

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (csaunders) [inline app/controllers/application_controller.rb:21]: Yeah it's probably best. Will require a bit of refactoring though.

**Comment by csaunders** [comment]
Updated and stuff

**Comment by carsonb** [comment]
:sheep: :8ball:

**Comment by carsonb** [comment]
Hey @csaunders
shiiippp iiiiiitttt

**Pull request merged**